templates/democracy/test/democracy.js

```diff
@@ -1,402 +1,316 @@
-                        const script2 = encodeCallScript([action2])
-                        const r = await tokenManager.forward(script2, { from: holder20 })
-                        voteId = getVoteId(r)
-                    })
-
-                    it('finance can not be accessed directly (without a vote)', async () => {
-                        try {
-                            await finance.newImmediatePayment(ETH, nonHolder, 2e16, "voting payment")
-                        } catch (err) {
-                            assert.equal(err.receipt.status, 0, "It should have thrown")
-                            return
-                        }
-                        assert.isFalse(true, "It should have thrown")
-                    })
-
-                    it('transfers funds if vote is approved', async () => {
-                        const receiverInitialBalance = await getBalance(nonHolder)
-                        //await logBalances(financeAddress, vaultAddress)
-                        await voting.vote(voteId, true, true, { from: holder29 })
-                        await voting.vote(voteId, false, true, { from: holder20 })
-                        //await timeTravel(votingTime + 1)
-                        await sleep(votingTime + 1)
-                        await voting.executeVote(voteId, {from: owner})
-                        //await logBalances(financeAddress, vaultAddress)
-                        assert.equal((await getBalance(nonHolder)).toString(), receiverInitialBalance.plus(payment).toString(), 'Receiver didn\'t get the payment')
-                    })
-                })
+            it('cannot execute vote if not support met', async () => {
+              await voting.vote(voteId, false, true, { from: holder29 })
+              await voting.vote(voteId, false, true, { from: holder20 })
+
+              await increaseTime(VOTING_TIME + 1)
+
+              await assertRevert(voting.executeVote.request(voteId, { from: owner }), 'VOTING_CAN_NOT_EXECUTE')
             })
+          })
         })
-    }
-
-    const logBalances = async(financeAddress, vaultAddress) => {
-        console.log('Owner ETH: ' + await getBalance(owner))
-        console.log('Finance ETH: ' + await getBalance(financeAddress))
-        console.log('Vault ETH: ' + await getBalance(vaultAddress))
-        console.log('Receiver ETH: ' + await getBalance(nonHolder))
-        console.log('-----------------')
-    }
 
-    const sleep = function(s) {
-        return new Promise(resolve => setTimeout(resolve, s*1000));
-    }
+        context('> Finance access', () => {
+          let voteId
+          const payment = new web3.BigNumber(2e16)
+
+          beforeEach('fund finance', async () => {
+            await finance.sendTransaction({ value: payment, from: owner })
+            const paymentAction = { to: finance.address, calldata: finance.contract.newImmediatePayment.getData(ETH, nonHolder, payment, 'voting payment') }
```

Comment:
Bit of a nitpick, technically this isn't really "funding" finance and could be moved to its own `beforeEach()`